ros_cross_compile/docker_client.py

```diff
@@ -104,14 +105,23 @@ def run_container(
             }
             for src, dest in volumes.items()
         }
-        logs = self._client.containers.run(
+        # Note that the `run` kwarg `stream` is not available
+        # in the version of dockerpy that we are using, so we must detach to live-stream logs
+        # Do not `remove` so that the container can be queried for its exit code after finishing
+        container = self._client.containers.run(
             image=image_name,
+            name=container_name,
             command=command,
             environment=environment,
             volumes=docker_volumes,
-            remove=True,
-            stream=True,
+            detach=True,
             network_mode='host',
         )
+        logs = container.logs(stream=True)
         for line in logs:
             logger.info(line.decode('utf-8').rstrip())
+        exit_code = container.wait()
```

Comment:
Do you want to move some of this cleanup code into a try...except to be exception safe?

Reply:
Ah, you mean just the `container.remove()` line, yeah I can put that inside an exception handler, good call.

Reply:
[```containers.wait```](https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.wait) can throw though, any reason not to catch in line 123?

Reply:
Added handler, and a test case